src/NexusMods.App.UI/Controls/ModInfo/ViewModFiles/ViewModFilesView.axaml.cs

```diff
@@ -0,0 +1,125 @@
+{
+    public ViewModFilesView()
+    {
+        InitializeComponent();
+        this.WhenActivated(_ =>
+        {
+            // Unleash the tree!
+            var source = CreateTreeSource(ViewModel!.Items);
+            source.SortBy(source.Columns[0], ListSortDirection.Ascending);
+            ModFilesTreeDataGrid.Source = source;
+            
+            // Hide Stack Panel
+            MultiLocationStackPanel.IsVisible = ViewModel.RootCount > 1;
+            SingleLocationStackPanel.IsVisible = ViewModel.RootCount <= 1;
+
+            SingleLocationText.Text = ViewModel.PrimaryRootLocation;
+            MultiLocationCountText.Text = $"{ViewModel.RootCount}";
+        });
+    }
+    
+    private static HierarchicalTreeDataGridSource<ModFileNode> CreateTreeSource(
+        ReadOnlyObservableCollection<ModFileNode> treeRoots)
+    {
+        return new HierarchicalTreeDataGridSource<ModFileNode>(treeRoots)
+        {
+            Columns =
+            {
+                new HierarchicalExpanderColumn<ModFileNode>(
+                    new TemplateColumn<ModFileNode>(
+                        Language.Helpers_GenerateHeader_NAME,
+                        new FuncDataTemplate<ModFileNode>((node, _) =>
+                            {
+                                // This should never be null but can be during rapid resize, due to
+                                // virtualization shenanigans. Think this is a control bug, but well, gotta work with what we have.
+                                // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+                                if (node == null)
+                                    return new Control();
+                                    
+                                // Very sus but it works, t r u s t.
+                                var view = new FileTreeNodeView();
+                                view!.DataContext = node.Item;
+                                
+                                // This is a 'hack' which allows us to receive events from the wrapping 'ModFileNode'
+                                // and transfer it into the child IFileTreeNodeViewModel.
+                                
+                                // Given that we may reuse the `FileTreeNodeView` for other views in the future, 
+                                // which may not necessarily need the wrapping `ModFileNode` to get info on when it's
+                                // expanded or not expanded, this does not seem unreasonable to do.
+                                ((IActivatableViewModel)node.Item).WhenActivated(d =>
+                                {
+                                    node.WhenAnyValue(x => x.IsExpanded)
+                                        .Subscribe(isExpanded => node.Item.OnExpanded(isExpanded))
+                                        .DisposeWith(d);
+                                });
```

Comment:
I agree, this can be eliminated by binding the view directly to the `ModFileNode` (`TreeNodeVM<IFileTreeNodeViewModel, GamePath>`).

The reason I did it this way however is because Al and Design wanted for the components used in this feature to be as reusable as possible. That means we may one day display `IFileTreeNodeViewModel` outside of the context of a `TreeDataGrid`, so I didn't really want the view to target a `TreeNodeVM`.

Although awkward, that's pretty much the reason it was done this way.

Reply:
Moved to #1048